### ISSUE TITLE
Order of array is changed on livewire request

### ISSUE THREAD

**Issue by intellow** [calebporzio/livewire#160]
**Summary:** _I have a key value pair ordered by value. After ANY livewire request, that gets re-ordered by key._

I have a variable that holds a list of key value pairs for a select dropdown. When I get the data from the database, I sort it by the value alphabetically.

When I make a livewire request, it returns the same array, but this time ordered by key instead of value, and basically scrambles the order of my select options. This happens both when I use a selectpicker and when I don't.

I have stripped everything down to bare bones. My component is just variables, a mount and the standard render function:

```
<?php

namespace App\Http\Livewire;

use App\Agency;
use Livewire\Component;

class AddParticipant extends Component
{
    public $agencies;
    public $programs;
    public $housingLocators;
    public $agency = null;
    public $caseManagers = [];
    public $caseManager = null;

    public $testing;

    /**
     * @param $agencies
     * @param $programs
     */
    public function mount($agencies, $programs)
    {
        $this->agencies = $agencies;
        $this->programs = $programs;
        $this->housingLocators = [];
    }

    public function render()
    {
        return view('livewire.add-participant');
    }
}
```

Then in the blade component:

```
<!-- Agency Form Input -->
        <div class="form-group col-12 col-md-6">
            {!! Form::select('agency', $agencies, $agency, [
                'class' => 'form-control',
                'title' => 'Agency',
                'required',
                'data-live-search' => 'true',
                'data-style' => 'select-with-transition',
                'data-size' => 5,
                'wire:model' => 'agency',
            ]) !!}
        </div>
```

I spit out the $agencies variable on the page just so you can see what happens when the livewire message is returned.

I can override this by using the `updated` hook in the class and making a call to the DB every time to get the ordered list, but obviously that's just a bandaid.

![2019-08-13_12-09-33 (1)](https://user-images.githubusercontent.com/40676515/62961988-863e1a00-bdc3-11e9-8267-ce5d6d61ed5f.gif)

**Comment by jmlallier** [comment]
@intellow I assume this is because some browsers JavaScript JSON parsing/encoding sorts arrays by key, whereas PHP's JSON parsing/encoding preserves order. To verify this is the case, can you run this on FireFox? FF should preserve the order, where Chrome does not. 

If this is the case, the only viable solution is to do as you mentioned and sort the data on the `updated` hook. Though you don't have to get it from the DB every time, you could use PHP to sort alphabetically by value using `asort($array)` (see [asort documentation](https://www.php.net/manual/en/function.asort.php)).

**Comment by intellow** [comment]
@jmlallier I just tested in Firefox and it's doing the same thing as Chrome. When I get some time I'll try to dig into the source and add a unit test to see if it's happening within Livewire.

**Comment by evergreenwebsystems** [comment]
@intellow did you have a chance to dig into the source yet? I am having the same problem. I created an example which is easy to spin up.

Use the Livewire example 'counter' component and add a static array of data into the ```mount``` function. The collect it, sort it and output it as an array.

```php
class Counter extends Component
{
    public $count = 0;
    public $labels = [];

    public function mount()
    {
        $labels =  [
            [
                'name' => 'Duplicate',
                'color' => 'bg-gray-500',
            ],
            [
                'name' => 'Enhancement',
                'color' => 'bg-blue-400',
            ],
            [
                'name' => 'Invalid',
                'color' => 'bg-yellow-500',
            ],
            [
                'name' => 'Question',
                'color' => 'bg-purple-500',
            ],
            [
                'name' => 'Wont fix',
                'color' => 'bg-green-500',
            ],
            [
                'name' => 'Bug',
                'color' => 'bg-red-500',
            ],
        ];

        $this->labels = collect($labels)->sortBy('name')->toArray();
    }

    public function increment()
    {
        $this->count++;
    }

    public function decrement()
    {
        $this->count--;
    }

    public function render()
    {
        return view('livewire.counter');
    }
}
```

Use the blade template for the 'Counter' example but add the new list of data.

```html
<div style="text-align: center">
    <button wire:click="increment">+</button>
    <h1>{{ $count }}</h1>
    <button wire:click="decrement">-</button>

    <div>
        <ul>
            @foreach ($labels as $label)
            <li>{{ $label['name'] }}</li>
            @endforeach
        </ul>
    </div>
</div>
```

The first time the page loads the data is correctly ordered, i.e. by name. 
![image](https://user-images.githubusercontent.com/8020291/63647518-9be80380-c722-11e9-9996-02c6819e7150.png)

If you then click the add button on the counter the counter updates, as expected, but also the data is re-ordered. This time it is sorted incorrectly, as Livewire appears to use the ```key``` rather than the defined ```name``` parameter.
![image](https://user-images.githubusercontent.com/8020291/63647521-a6a29880-c722-11e9-87aa-f54d7730b72d.png)

If I ```dd($this->labels)``` in the component the order is correct always, so it seems that Livewire is sorting the data in the DOM maybe?

**Comment by intellow** [comment]
I haven't had a chance to look in the source on this yet. Keep in mind you can override this behavior by sorting in an `updated` function. It's not a great solution, but does work as a workaround.

**Comment by 2kodes** [comment]
> I haven't had a chance to look in the source on this yet. Keep in mind you can override this behavior by sorting in an `updated` function. It's not a great solution, but does work as a workaround.

the sorting can also be done in the render or hydrate lifecycle function ..

**Comment by calebporzio** [comment]
Should be fixed in v0.3.3: https://github.com/calebporzio/livewire/releases/tag/v0.3.3

**Comment by fontenele** [comment]
Fixed setting $this->labels in render() method
